### ISSUE TITLE
Warnings introduced by #31 restartPolicy change

### ISSUE THREAD

**Issue by mleklund** [stafftastic/jitsu-chart#36]
Warnings were introduced in the change #31 since spec.restartPolicy does not exist for deployments.  You likely want .spec.template.spec.restartPolicy which [already defaults to Always](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#pod-template), so I am not sure this fixed anything for you.

**Comment by rodfsoares** [comment]
in my case, I'm actually getting errors, because our CI pipeline uses a Helm linting tool that fails for unrecognized parameters 😅, and throws an error:
```
Deployment in version "v1" cannot be handled as a Deployment: strict decoding error: unknown field "spec.restartPolicy"
```

similar to this one:
https://stackoverflow.com/questions/57233686/error-when-creating-deployment-yaml-deployment-in-version-v1-cannot-be-hand

could we possibly remove this incorrect `restartPolicy`?

**Comment by rodfsoares** [comment]
opened a PR for your consideration 🙏 

although this issue might translate to a **warning** in some cases, in others where there is a stricter Helm linting process, it translates into an **error** that causes Helm templating to fail entirely.

**Comment by echozio** [comment]
I merged the PR and released v1.8.4. The reason I did this was that I saw some pods got stuck as "Completed" when they crashed but didn't return a non-zero status. Since Always is the default it must have been caused by something else and my implementation was nonetheless rushed and wrong.
